core/modules/qdisp/testQDisp.cc

```diff
@@ -253,53 +163,89 @@ void timeoutFunc(util::Flag<bool>& flagDone, int millisecs) {
 }
 
+// The following sets up the environment to do a test and is modeled after
+// ccontrol::UserQuery::submit() (note that we cannot reuse an executive).
+//
+class SetupTest {
+    public:
+    std::string qrMsg;
+    std::string str;
+    qdisp::Executive::Config::Ptr conf;
+    std::shared_ptr<qdisp::MessageStore> ms;
+    qdisp::LargeResultMgr::Ptr lgResMgr;
+    qdisp::Executive::Ptr ex;
+    std::shared_ptr<qdisp::JobQuery> jqTest; // used only when needed
+
+    SetupTest(const char *request) {
+         qrMsg = request;
+         qdisp::XrdSsiServiceMock::Reset();
+         str = qdisp::Executive::Config::getMockStr();
+         conf = std::make_shared<qdisp::Executive::Config>(str);
+         ms = std::make_shared<qdisp::MessageStore>();
+         lgResMgr = std::make_shared<qdisp::LargeResultMgr>();
+         ex = qdisp::Executive::newExecutive(conf, ms, lgResMgr);
+    }
+    ~SetupTest() {}
+};
+
 BOOST_AUTO_TEST_SUITE(Suite)
 
+    // Variables for all subsequent tests. Note that all tests verify that the
+    // resource object for all chuncks has been properly constructed. We use
+    // the same chunkID for all tests (see setRName() below).
+    //
+    SequentialInt sequence(0);
+    int chunkId = 1234, millisInt = 200;
+
 BOOST_AUTO_TEST_CASE(Executive) {
-    _log.setLevel(LOG_LVL_DEBUG);
-    LOGS_DEBUG("Executive test 1");
+    // Variables for all executive sub-tests. Note that all executive tests
+    // are full roundtrip tests. So, if these succeed then it's likely all
+    // other query tests will succeed. So, much of this is redundant.
+    //
     util::Flag<bool> done(false);
-    // Modeled after ccontrol::UserQuery::submit()
-    std::string str = qdisp::Executive::Config::getMockStr();
-    qdisp::Executive::Config::Ptr conf = std::make_shared<qdisp::Executive::Config>(str);
-    std::shared_ptr<qdisp::MessageStore> ms = std::make_shared<qdisp::MessageStore>();
-    qdisp::LargeResultMgr::Ptr lgResMgr = std::make_shared<qdisp::LargeResultMgr>();
-    qdisp::Executive::Ptr ex = qdisp::Executive::newExecutive(conf, ms, lgResMgr);
-    SequentialInt sequence(0);
-    SequentialInt chunkId(1234);
     int jobs = 0;
-    // test single instance
-    int millisInt = 200;
+    _log.setLevel(LOG_LVL_DEBUG); // Ugly but boost test suite forces this
     std::thread timeoutT(&timeoutFunc, std::ref(done), millisInt*10);
-    std::string millis(boost::lexical_cast<std::string>(millisInt));
-    ++jobs;
-    executiveTest(ex, sequence, chunkId, millis, 1);
-    LOGS_DEBUG("jobs=" << jobs);
-    ex->join();
-    BOOST_CHECK(ex->getEmpty() == true);
-
-    // test adding 4 jobs
-    LOGS_DEBUG("Executive test 2");
-    executiveTest(ex, sequence, chunkId, millis, 4);
+    qdisp::XrdSsiServiceMock::setRName((std::string)"/chk/Mock/1234");
+
+    // Test single instance
+   {LOGS_DEBUG("Executive single query test");
+    SetupTest tEnv("respdata");
+    tEnv.jqTest = executiveTest(tEnv.ex, sequence, chunkId, tEnv.qrMsg, 1);
```

Comment:
I'd format this with proper indentation to make block structure more obvious:

    {
        LOGS_DEBUG("Executive single query test");
        SetupTest tEnv("respdata");
        // ...
    }